blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -137,10 +138,13 @@ class RunDetails extends Component {
     afterClose = () => {
         const { router, params } = this.context;
         if (this.opener) {
-            router.goBack();
+            // step back in the history to the item that's prior to initial load of RunDetails
+            const offsetToInitialRoute = this.initialHistoryLength - history.length;
+            router.go(offsetToInitialRoute - 1);
```

Comment:
I'm not fully up to speed on all the navigation flows involving in the editor. Above, are you describing the following flow?

1. User opens modal for a run,
2. User clicks "edit" to navigate to editor,
3. User presses back button

Need to make sure I test more wacky scenarios with this flow. After talking to @michaelneale I'm also going to file a ticket for us to consider the design of this modal and how it works. These interactions are causing a tremendous amount of development and testing pain.